hrabsence/CRM/HRAbsence/Form/AbsenceRequest.php

```diff
@@ -517,320 +541,410 @@ static function formRule($fields, $files, $self) {
+      $activityLeavesParam['activity_date_time'] = $date;
+      $activityLeavesParam['duration'] = $duration['duration'];
+      $activityLeavesParam['status_id'] = $duration['approval'];
+      civicrm_api3('Activity', 'create', $activityLeavesParam);
+    }
+  }
+
+  /**
+   * Creates array of e-mail destinataries according to contact data.
+   * 
+   * @param array $sendTemplateParams
+   *   Paraameters of template to be sent as an e-mail message
+   * 
+   * @return array
+   *   Array of destinataries
+   */
+  private function buildMailDestinatariesParams($sendTemplateParams) {
+    $mailprm = [];
+
+    // Add target contact
+    $mailprm[$this->_targetContactID]['display_name'] = $this->_targetContact['display_name'];
+    $mailprm[$this->_targetContactID]['email'] = $this->_targetContact['email'];
+
+    // Add managers
+    $managerContactResult = [];
+    if (($this->_action & (CRM_Core_Action::ADD) || $sendTemplateParams['sendMail']) && !empty($this->_managerContactID)) {
+      foreach ($this->_managerContactID as $val) {
+        $managerContactResult = civicrm_api3('contact', 'get', [
+          'id' => $val,
+        ]);
+        if (!empty($val) && !empty($managerContactResult['values'])) {
+          $mailprm[$val]['display_name'] = $managerContactResult['values'][$val]['display_name'];
+          $mailprm[$val]['email'] = $managerContactResult['values'][$val]['email'];
         }
-        self::sendAbsenceMail($mailprm, $sendTemplateParams);
       }
     }
-    else {
-      if (CRM_Utils_Request::retrieve('aid', 'Positive', $this)) {
-        $activityIDs = CRM_Utils_Request::retrieve('aid', 'Positive', $this);
-      }
 
-      if (array_key_exists('_qf_AbsenceRequest_done_cancelabsence', $submitValues)) {
-        $statusId = CRM_Utils_Array::key('Cancelled', $activityStatus);
-        $statusMsg = ts('Absence(s) have been Cancelled');
-      }
-      elseif (array_key_exists('_qf_AbsenceRequest_done_cancel', $submitValues)) {
-        $session->pushUserContext(CRM_Utils_System::url('civicrm/absences', "reset=1&cid={$this->_targetContactID}", false, 'hrabsence/list'));
-      }
-      civicrm_api3('Activity', 'create', array(
+    return $mailprm;
+  }
+
+  /**
+   * Updates acivities to given status for current leave request.
+   * 
+   * @param type $status
+   *   Status to which activities should be set
+   * 
+   * @return array
+   *   Data of main activity for curent leave request
+   */
+  private function updateActivitiesStatus($status) {
+    $statusId = CRM_Utils_Array::key($status, $this->_activityStatuses);
+    $result = civicrm_api3('Activity', 'create', 
+      $activityParam = [
+        'sequential' => 1,
         'id' => $this->_activityId,
         'activity_type_id' => $this->_activityTypeID,
         'status_id' => $statusId
+      ]
+    );
+    $subact = civicrm_api3('Activity', 'get', ['return' => "id", 'source_record_id' => $result['id']]);
```

Comment:
also instead of naming the variable $subact .. change it to $subActivity .. it is easier to read like that